rfd/0073-discover.md

```diff
@@ -0,0 +1,203 @@
+   cloud resources.
+
+Let's explore in more detail what the flow for both of these user personas
+would look like.
+
+## Day 1
+
+Day 1 users should be greeted by a wizard-style dialog upon logging into the
+web UI of a cluster that does not have any connected resources. The wizard
+will guide the user through the flow of connecting their first resource.
+
+Teleport web UI already provides some of the building blocks for the wizard in
+the form of "Add server", "Add database", etc. pop-up dialogs but their
+instructions are not friendly for newcomers and make it almost impossible to
+use successfully without referring to the documentation.
+
+Instead of separate dialogs, Teleport Discover wizard will provide a unified
+experience to enable the flow described below.
+
+Users should be able to navigate between the dialog steps back and forth to
+connect multiple resource if needed. The unified "Add resource" wizard should
+also be available and prominently visible in the web UI allowing users to go
+through the same flow in a non-empty cluster.
+
+### Step 1. Gather information
+
+Ask the user what type of resource they would like to connect: an SSH server,
+a database, an application, etc.
+
+Then ask the user where the resource is located: self-hosted or AWS. For
+databases, also present the supported protocol options. For self-hosted:
+PostgreSQL, MySQL, MongoDB, Redis, SQL Server. For AWS: RDS PostgreSQL, RDS
+MySQL, RDS SQL Server, Elasticache, MemoryDB.
+
+### Step 2. Install Teleport
+
+For SSH, instruct the user to download and install `teleport` binary on the
+server they're intending to connect. Use tabs to display per-distro install
+instructions similar to [the docs](https://goteleport.com/docs/server-access/getting-started/#step-14-install-teleport-on-your-linux-host).
+
+For a database, display the same instructions making it clear that Teleport
+should be installed on a node that can reach the database.
+
+### Step 3. Configure node
+
+For SSH, this step is a no-op as it doesn't need any additional configuration
+so users will proceed to starting an agent at step 4.
+
+For databases, this step will inform the user of any additional configuration
+they may need in order to get database access to work. It will depend on the
+database type and where the database is hosted.
+
+For self-hosted databases, in the initial version the wizard will display the
+links to the respective sections of the documentation guides for preparing
+the node (e.g. joining to AD domain for SQL Server) and show the commands for
+creating a database user.
+
+For AWS databases that use IAM authentication (PostgreSQL, MySQL), the wizard
+will additionally display an appropriate `teleport db configure bootstrap`
+command for the user to run which will configure IAM permissions.
+
+### Step 4. Start agent
+
+For SSH, display the command for the user to run on their node, showing either
+"Automatic" or "AWS" commands from the existing "Add server" dialog depending
```

Comment:
nit: let's please change Automatic to something else - when I saw it after a while, I got confused - is AWS not automatic? What's the difference?